hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -175,7 +177,8 @@ def output(s):
                     rule, data = cd.draw(machine._rules_strategy)
                 draw_label = f"generate:rule:{rule.function.__name__}"
                 cd.draw_times.setdefault(draw_label, 0.0)
-                cd.draw_times[draw_label] += perf_counter() - start_draw
+                in_gctime = gc_cumulative_time() - start_gc
+                cd.draw_times[draw_label] += perf_counter() - start_draw - in_gctime
```

Comment:
We'll need to do this for the stateful-specific `cd._stateful_run_times` below too.